app/controllers/auth_controller.rb

```diff
@@ -4,11 +4,47 @@ class AuthController < ApplicationController
   skip_before_filter :verify_authenticity_token, :only => [:access_token]
 
   def verify_logged_in
-    if current_user.nil?
-      session[:sso_callback_params] = params
-      session[:sso_application] = application
-      redirect_to auth_login_path
+
+    action = params[:action]
+
+    if action == 'oauth_authorize'
```

Comment:
Instead of adding the logic here to limit this code to the `oauth_authorize` action. It seems better to just move it into the oauth_authorize method.  And then change the before filter to: 
`before_filter :verify_logged_in, :except => [:access_token, :login, oauth_authorize]`